gradio/documentation.py

```diff
@@ -16,18 +17,41 @@ def set_documentation_group(m):
         classes_to_document[m] = []
 
 
-def document(*fns):
+def extract_instance_attr_doc(cls, attr):
+    code = inspect.getsource(cls.__init__)
+    lines = [line.strip() for line in code.split("\n")]
+    found_attr = False
+    i = 0
+    for i, line in enumerate(lines):
+        if line.startswith("self." + attr):
+            found_attr = True
+            break
+    assert found_attr, f"Could not find {attr} in {cls.__name__}"
+    start_line = lines.index('"""', i)
+    assert start_line != -1, f"Could not find docstring for {attr} in {cls.__name__}"
```

Comment:
How could `start_line` or `end_line` ever be -1?

Reply:
If `lines.index('"""', i)` fails, a `ValueError` is thrown, so I don't think you need the assertion errors

Reply:
-1 meaning it could not find the docstring start quotes

Reply:
That's not how `.index()` works though

Reply:
Oh you're right, I'm thinking of js